[D2] Icon 구현 및 Preview 작성

szzang-kr opened the pull request:
# 개요
- 🔗  이슈링크 : #8 


# 변경사항

## 작업 내용
<!-- 작업한 코드/UI에 대한 설명을 작성합니다. -->
* DesignSystem에 앱 전체에서 사용될 Icon 등록 및 Image의 Extension으로 사용할 수 있도록 기능 구현
* 아래와 같이 사용할 수 있습니다.
```swift 
import DesignSystemKit

Image( {아이콘 namespace} )

Image(Icon.Chevron.upBig)
Image(Icon.Chevron.upMedium)

Image(Icon.Checkmark.trueFill24)
Image(Icon.Checkmark.falseLine20)
```


### 스크린샷
<!-- 작업 전/후 UI 수정이 있을 경우 스크린샷을 첨부합니다. -->
스크린샷 1|스크린샷2|스크린샷3
---|---|---
![Simulator Screenshot - iPhone 14 Pro - 2023-05-29 at 14 27 55](https://github.com/mash-up-kr/weQuiz-iOS/assets/39300449/e2644afb-fbc2-45ed-912d-f0dc058e97b8)|![Simulator Screenshot - iPhone 14 Pro - 2023-05-29 at 14 27 59](https://github.com/mash-up-kr/weQuiz-iOS/assets/39300449/cdfe5552-b3da-49f1-b166-963945f6af41)|![Simulator Screenshot - iPhone 14 Pro - 2023-05-29 at 14 28 03](https://github.com/mash-up-kr/weQuiz-iOS/assets/39300449/af5bb655-5645-48b3-b5d4-f07600e0a299)

szzang-kr committed:
:bulb: Icon 구현 및 Preview 작성(#8)

DesignSystem에 앱 전체에서 사용될 Icon 등록 및 Image의 Extension으로 사용할 수 있도록 기능 구현

sohy1018 commented:
```
Image.icon(.Chevron.upBig)
Image.icon(Icon.Chevron.upMedium)

Image.icon(.Checkmark.trueFill24)
Image.icon(.Checkmark.falseLine20)
```

`icon`이라는 단어가 중복되니 이런 식으로 사용할 수 있으면 좋겠습니다~

dittoo8 reviewed (changes requested):
```
Image.icon(.Chevron.upBig)
Image.icon(Icon.Chevron.upMedium)

Image.icon(.Checkmark.trueFill24)
Image.icon(.Checkmark.falseLine20)
```

`icon`이라는 단어가 중복되니 이런 식으로 사용할 수 있으면 좋겠습니다~

szzang-kr committed:
: recycle: Icon 리뷰 적용(#8)

Icon의 DesignSystemKit Prefix 제거, Image의 static func 대신 initializer를 사용할 수 있도록 변경

szzang-kr commented:
> ```
> Image.icon(.Chevron.upBig)
> Image.icon(Icon.Chevron.upMedium)
> 
> Image.icon(.Checkmark.trueFill24)
> Image.icon(.Checkmark.falseLine20)
> ```
> 
> 
>      
>   
> 
> `icon`이라는 단어가 중복되니 이런 식으로 사용할 수 있으면 좋겠습니다~

https://github.com/mash-up-kr/weQuiz-iOS/pull/15/commits/f6aab164dfe0f408e4b1e2e9cc00447c7f7f2fef 커밋에서 반영완료!

아래와 같이 사용가능
```
import DesignSystemKit

Image( {아이콘 namespace} )

Image(Icon.Chevron.upBig)
Image(Icon.Chevron.upMedium)

Image(Icon.Checkmark.trueFill24)
Image(Icon.Checkmark.falseLine20)
```
